source.go

```diff
@@ -20,12 +20,36 @@ var (
 	_ (Source) = (*MultipleSource)(nil)
 )
 
+type SourceOptFunc func(*sourceOpt)
```

Comment:
Nitpick(naming): I would generally opt for naming like `SourceOption`, where `sourceOpt` becomes `sourceOptions` (ie. the singular Option is the function operating on the plural Options struct). Feels a little more idiomatic.

Reply:
I like that a bit better as well actually, I have renamed it and moved the options into their own file.